pkg/block/metadata/meta.go

```diff
@@ -54,14 +54,30 @@ type Meta struct {
 
 // Thanos holds block meta information specific to Thanos.
 type Thanos struct {
+	// Version of Thanos meta file. If none specified, 1 is assumed (since first version did not have explicit version specified).
+	Version int `json:"version,omitempty"`
+
 	Labels     map[string]string `json:"labels"`
 	Downsample ThanosDownsample  `json:"downsample"`
 
 	// Source is a real upload source of the block.
 	Source SourceType `json:"source"`
 
 	// List of segment files (in chunks directory), in sorted order. Optional.
+	// Deprecated. Use Files instead.
 	SegmentFiles []string `json:"segment_files,omitempty"`
+
+	// File is a sorted (by rel path) list of all files in block directory of this block known to TSDB.
+	// Sorted by relative path.
+	// Useful to avoid API call to get size of each file, as well as for debugging purposes.
+	// Optional, added in v0.17.0.
+	Files []File `json:"files,omitempty"`
+}
+
+type File struct {
+	RelPath string `json:"rel_path"`
+	// SizeBytes is optional (e.g meta.json does not show size).
+	SizeBytes int64 `json:"size_bytes,omitempty"`
```

Comment:
We are using `camelCase` in all other parts of the `meta.json` so I think it makes sense to have these properties in camelcase as well.

I noticed that the `segment_files` is also not in camel case, I am not sure how much impact it will have on existing installations if we change it to `segmentFiles`.

Reply:
> I noticed that the segment_files is also not in camel case, I am not sure how much impact it will have on existing installations if we change it to segmentFiles.

It will be :hankey: =D